File: Practice Solutions/9-12-19/gophers.cpp
#include<bits/stdc++.h>

using namespace std;

int main(){
    cin.tie(NULL);
    ios_base::sync_with_stdio(false);

    int T, N, M;
    cin >> T >> N >> M;

    for(int test = 0; test < T; test++){
        int *sol  = new int[M];

        int guesses [7] = {18,17,16,15,14,13,11};

        for(int guess = 0; guess < 7; guess++){
            for(int i = 0; i < 18; i++)         
                cout << guesses[guess] << ' ';
            cout << endl;

            int count = 0;
            for(int i = 0; i < 18; i++){
                int temp;
                cin >> temp;
                count += temp;
            }

            count %= guesses[guess];

            int insert = 0;
            while(insert * guesses[guess] + count <= M){
                sol[insert * guesses[guess] + count] += 1;
                insert += 1;
            }
        }
        int output = -1;
        for(int i = 0; i <= M; i++){
            if(sol[i] == 7){
                output = i;
                break;
            }
        }
        cout << output << endl;

        cin >> output;
        if(output == -1)
            break;
    }
}

// https://codingcompetitions.withgoogle.com/codejam/round/0000000000051635/0000000000104f1a

File: Practice Solutions/9-12-19/gophers.py
T, N, M = list(map(int, input().split()))
for test in range(0,T): 

    guesses = [18,17,16,15,14,13,11]
    mods = []
    
    for guess in guesses:
        
        print((str(guess)+ " ")*18)
        
        output = list(map(int, input().split()))

        count = 0
        for el in output:
            count += el

        count %= guess
        mods.append(count)

    for i in range(M + 1):
        for mod, guess in zip(mods,guesses):
            if i % guess != mod:
                break
        else:
            print(i)
            break
    else:
        print(-1)

    res = int(input())
    if res == -1:
        break